syntax = "proto3";

package wal_manager;

service WalManager {
   rpc process_query(query_req) returns (query_res) {}
}

message query_req {
   string key = 1;
   string type = 2;
   string value = 3;
}

message Column {
   string column = 1;
   string datatype = 2;
   string ctype = 3;
   int32 clen = 4;
   int32 precision = 5;
   string value = 6;
}

message Wal {
   string table_name = 1;
   string index_val = 2;
   repeated Column column_list = 3;
}
   
message query_res {
   string key = 1;
   repeated Wal wal_list = 2; 
}

